pkg/apply/apply.go

```diff
@@ -854,13 +860,22 @@ func (t *TopicApplier) updatePlacementRunner(
 		)
 	}
 
+	numRounds := (len(assignmentsToUpdate) + batchSize - 1) / batchSize // Ceil() with integer math
+	round := 0
```

Comment:
What do you think about refactoring this for loop to use the round and numRounds variables? It is a bit confusing how we are tracking the iteration count in two ways right now

Reply:
I didn't really like having two separate counters but I tried a couple of variations and none seem any better. If I use `round` as the counter, I have to calculate `i` from it (eg `i := round * batchSize`) anyway.

If I keep the loop based on `i`, but calculate `rounds` based on `i` (`round := i/batchSize+1`), it doesn't seem than just independently incrementing it.

Any other ideas?

Reply:
Ah, yeah maybe I am off base here. Does initializing both iterables and incrementing them in the for loop initialize like this seem any cleaner?

```go
for i, round := 0, 0; i < len(assignmentsToUpdate); i += batchSize, round += 1
```

Reply:
Hmm, I've never tried that before. It's unusual and a little harder to read but I like that it explicitly pins the round counter to the `for` statement 👍